libkbfs/folder_branch_ops.go

```diff
@@ -3320,8 +3320,10 @@ func (fbo *folderBranchOps) Read(
 			return err
 		}
 
+		// Read using the `file` Node, not `filePath`, since the path
+		// could change until we take `blockLock` for reading.
```

Comment:
Nah -- there we're just checking the path names, not the BlockPointers.  I suppose if someone renames the .git directory to something else right in the middle of a read, then the special git behavior may or may not be enabled, but that seems ok.  As long as the right file contents is still read.